server/core/src/reServerLib.cpp

```diff
@@ -409,7 +409,15 @@ runQueuedRuleExec( rcComm_t *rcComm, reExec_t *reExec,
     int thrInx;
 
     inx = -1;
-    while ( time( NULL ) <= endTime && ( thrInx = allocReThr( reExec ) ) >= 0 ) {
+    while ( time( NULL ) <= endTime ) {
+	if ( ( thrInx = allocReThr( reExec ) ) < 0 ) {
+	    if ( reExec->doFork == 0 || waitAndFreeReThr( rcComm, reExec ) < 0 ) {
+		break;
+	    }
```

Comment:
At this point there is no reExecProc available to run a delayed rule. Break out of the loop immediately when running single-threaded.

Otherwise, wait for a child process to exit; if there is any kind of error, managing subprocesses has failed and we won't assume they can safely be restarted.